### DISCUSSION TITLE
Problem on align two tables side by side

### DISCUSSION THREAD

**Discussion by elsenthil123** [dompdf/dompdf#2726] [Q&A]
Hi, 

I am trying to align two tables side by side.

My table rows in foreach loop. some times my table rows go to the 2nd or 3rd page also.  If the values go to the 2nd or 3rd page, my right side table starts from the 2nd or 3rd page.

If the rows end on a single page both right and left side tables are working properly.

My pdf output files are attached bellow

[2 page readings.pdf](https://github.com/dompdf/dompdf/files/7824317/2.page.readings.pdf)
[Single page readings.pdf](https://github.com/dompdf/dompdf/files/7824321/Single.page.readings.pdf)

My code is 

```
 <main>
  <header>
		<table>           
			
			<th1 ><img src='".$logo."' alt='".$logo."' style='width:180px;height:50px;' /><br /></th1>
			<th2 ></th2>           
			
			</table>
        </header>
 <h2 >Ref: SSI/$bnch/$date2/$id</h2>
 
<table width=\"100.5%\" border=\"0\" cellspacing=\"0\" cellpadding=\"3\" style=\"border:2px solid #000000\">
 <tr>
 <td class='head1'><b>Calibration Report for Aggregate Weigher</b></td>

 </tr>
 </table>
 
<table width=\"100.5%\" border= 2px solid #000000 cellspacing=\"0\" cellpadding=\"0\" style=\"border:2px solid #000000\" >
<thead>
<tr>
<td class='head2'><b>Ascending Order  </b></td>
<td class='head2'><b>Descending Order</b></td>
</tr>
</thead>
</table>


<table width=\"50%\"  border=\"1\" cellspacing=\"0\" cellpadding=\"1\" align=\"left\" style=\"border:2px solid #000000\"; >
<thead>
    <tr>
        <th class='head2'><b>Actual <br> Reading</b></th>
        <th class='head2'><b>Meter<br> Reading</b></th>
        <th class='head2'><b>Error</b></th>
</tr>
</thead>
";
      	$json_var = json_decode($aggregateascending,true);			
        foreach($json_var as $key=>$value){
	    $ascending = explode(',',$value);
            
$entry = "   
<tbody>
		<tr>
		    <td class='head2'>$ascending[0]</td>
			<td class='head2'>$ascending[1]</td>
			<td class='head2'>$ascending[2]</td>
		</tr>
";
$html .=$entry;
		  }
$html1 = "
 </tbody>
        </table>

<table width=\"50%\"  border=\"1\" cellspacing=\"0\" cellpadding=\"1\" align=\"right\" style=\"border:2px solid #000000\"; >
<thead>
    <tr>
        <th class='head2'><b>Actual <br> Reading</b></th>
        <th class='head2'><b>Meter<br> Reading</b></th>
        <th class='head2'><b>Error</b></th>
</tr>
</thead>
";
$html .=$html1;
      	$json_var = json_decode($aggregatedescending,true);			
        foreach($json_var as $key=>$value){
	    $ascending1 = explode(',',$value);
            
$entry = "   
<tbody>
		<tr>
		    <td class='head2'>$ascending1[0]</td>
			<td class='head2'>$ascending1[1]</td>
			<td class='head2'>$ascending1[2]</td>
		</tr>
";
$html .=$entry;
		  }
$html1 = "
 </tbody>
        </table>
        
  <footer>
  <h2><picture>
      <img src='".$signature."' alt='".$signature."' style='width:150px;height:100px;' />
</picture></h2>
</footer>
        
 </main>
```

My full code is attached here for your reference
[pdf2.txt](https://github.com/dompdf/dompdf/files/7824331/pdf2.txt)

Ana anyone please correct me where i am doing mistake on my code

**Comment by bsweeney** [comment]
This is an unfortunate result of how Dompdf renders content, which is sequentially and elements start on the current page. So even though this layout should conceptually support side-by-side rendering it's just not handled very well.

If one of the tables is always under a page you should place that one first in the document. Then it would render as expected.

One update I hope to see in the future is for floated content (e.g. your aligned tables) to pause rendering at the end of a page so that in-flow content can render. Making that change would allow us to support your particular layout.

**Reply by elsenthil123** [reply]
Hi 

Can you please explain me on my code that how to do this

**Reply by bsweeney** [reply]
Looking at the actual output again my suggestion is probably not viable. In the multi-page scenario both tables are longer than a single page so regardless of which table renders first the result will be one table starting on a separate page. I think right now the only way you can make this work with your current layout would be to manually split the table after a certain number of rows. You _could_ look at creating a DIV-based tabular structure to accommodate the rendering limitations, but it would be fairly complicated to create and highly inefficient.

For a layout like this, you might be better off with something like wkhtmltopdf.

**Reply by elsenthil123** [reply]
Thanks for your reply. I will try your suggestion
